File: Newbee.UI/src/app/core/models/company/responses/company-response.ts
import { ActiveCodeResponse } from "../../active-code/response/active-code-response";
import { BranchResponse } from "../../branch/responses/branch-response";
import { ManagerResponse } from "../../manager/responses/manager-response";

export interface CompanyResponse {
  id : number;
  name : string;
  rin : string;
  activeCodes : ActiveCodeResponse[];
  branches : BranchResponse[];
  managers : ManagerResponse[];
}

File: Newbee.BE/NOTE.Solutions/NOTE.Solutions.BLL/Contracts/Company/Responses/CompanyResponse.cs
using NOTE.Solutions.BLL.Contracts.Manager.Responses;
using NOTE.Solutions.Entities.Entities.Company;

namespace NOTE.Solutions.BLL.Contracts.Company.Responses;

public class CompanyResponse
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string RIN { get; set; } = string.Empty;
    public List<ActiveCodeResponse> ActiveCodes { get; set; } = [];
    public List<BranchResponse> Branches { get; set; } = default!;
    public List<ManagerResponse> Managers { get; set; } = [];
}
